### PR TITLE
Enable 'retirement' of paypaldp/paypalwpp

### LINKED ISSUE #6126: Payment Methods: Adding switch to enable admin handling only
### Discussed in https://github.com/zencart/zencart/discussions/6088

<div type='discussions-op-text'>

<sup>Originally posted by **lat9** December 15, 2023</sup>
If a site's currently using payment-module `A`, but decides to switch to payment-module `B` ... payment-module `A` is disabled so that new orders can't be placed.  Unfortunately, that disablement results in payment-module `A` being disabled both storefront and admin.

This could pose problems for a store if they've still got orders that need to be either captured, refunded or voided that were placed using payment-module `A`.

Perhaps payment-modules should provide a tri-state "Enable this" setting to disable both storefront/admin, enable both storefront/admin or disable storefront-only?</div>

### PR DESCRIPTION
**Pull request by lat9**
Fixes #6126

### CODE REVIEW

**Comment by scottcwilson** [comment]
Does it make sense to wrap the "Retired" update in a check so we don't do it every time?

**Comment by lat9** [comment]
No.  In order to see if the "Retired" option has been applied, you'd need to perform a database query.  It's simpler to just perform the update, which would need to be done anyway.

**Comment by lat9** [comment]
... although it makes sense to limit that forced update to the admin's **_Modules :: Payment_** page; I'll get those updated.

File: includes/modules/payment/paypalwpp.php [hunk 1]
```diff
@@ -177,6 +177,8 @@ function __construct() {
 
       if (null === $this->sort_order) return false;
 
+      $this->enabled = (MODULE_PAYMENT_PAYPALWPP_STATUS === 'True' || (IS_ADMIN_FLAG === true && MODULE_PAYMENT_PAYPALWPP_STATUS === 'Retired'));
```
Comment (torvista) [inline includes/modules/payment/paypalwpp.php:180]: $this->enabled is now inside the IS_ADMIN clause: this payment option will never show on the shopfront.

Comment (lat9) [reply includes/modules/payment/paypalwpp.php:180]: True, that.  I'll get this updated later today.

Comment (lat9) [reply includes/modules/payment/paypalwpp.php:180]: Sigh, the `paypaldp` has issues as well.  Both modules have some additional checks if enabled to set their title/description.

Comment (drbyte) [reply includes/modules/payment/paypalwpp.php:180]: @lat9 wouldn't just moving that $this->enabled check back to its original line be enough? (er, well, probably also restore the original `defined()` check as well).

I'm not seeing why it needed to be relocated.

Comment (lat9) [reply includes/modules/payment/paypalwpp.php:180]: That's me over-thinking again; you're right.

**Pull request merged**